gossip/mempool.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+// Tx is a transaction that can be gossiped across the network.
+type Tx interface {
+	// ID returns the unique id of this transaction
+	ID() ids.ID
+	// Marshal returns the byte representation of this transaction
+	Marshal() ([]byte, error)
+	// Unmarshal deserializes the provided bytes in-place
+	Unmarshal(b []byte) error
+}
+
+// Mempool holds pending transactions
+type Mempool[T Tx] interface {
```

Comment:
super nit: maybe rename `Mempool` to something like `TxContainer`? I understand this is the interface for the backend gossip needs to work

Reply:
Keeping this as mempool since I think it's a more idiomatic name